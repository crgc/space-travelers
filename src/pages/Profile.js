import { useSelector } from 'react-redux';
import MyItems from '../components/profile/MyItems';

const Profile = () => {
  const storeMissions = useSelector((state) => state.missionsReducer);
  const rockets = useSelector((state) => state.rocketsReducer);

  const missionsReverved = storeMissions.filter(((mission) => mission.reserved))
    .map((mission) => ({
      id: mission.mission_id,
      name: mission.mission_name,
    }));

  const rocketsReserved = rockets.filter((rocket) => rocket.reserved);
  return (
    <div className="d-flex flex-row justify-content-around mt-3 profile-container">
      <MyItems title="My Missions" items={missionsReverved} />
      <MyItems title="My Rockets" items={rocketsReserved} />
    </div>
  );
};

export default Profile;
