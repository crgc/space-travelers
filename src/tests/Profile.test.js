import React from 'react';
import renderer from 'react-test-renderer';
import Profile from '../pages/Profile'
import { Provider } from 'react-redux';
import store from '../redux/configureStore';

it('Profile page renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}><Profile /></Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});