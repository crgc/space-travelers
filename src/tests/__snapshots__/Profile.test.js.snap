// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page renders correctly 1`] = `
<div
  className="d-flex flex-row justify-content-around mt-3 profile-container"
>
  <div
    className="list-group-container"
  >
    <h3>
      My Missions
    </h3>
    <div
      className="mt-3 list-group"
      onKeyDown={[Function]}
    />
  </div>
  <div
    className="list-group-container"
  >
    <h3>
      My Rockets
    </h3>
    <div
      className="mt-3 list-group"
      onKeyDown={[Function]}
    />
  </div>
</div>
`;
